### PR TITLE
add autoIncrement implement for #5266

### LINKED ISSUE #5266: 使用"gorm.io/gen" 生成的结构体，在后来使用 "gorm.io/gorm" 还原数据库表结构时，表中的PrimaryKey缺少AutoIncrement
## 使用"gorm.io/gen" 生成的结构体，在后来使用 "gorm.io/gorm" 还原数据库表结构时，表中的PrimaryKey缺少AutoIncrement

1、使用Mysql 建立如下数据库example, 建立如下表people：
```sql
SET NAMES utf8mb4;
SET FOREIGN_KEY_CHECKS = 0;

-- ----------------------------
-- Table structure for people
-- ----------------------------
DROP TABLE IF EXISTS `people`;
CREATE TABLE `people`  (
  `id` int(20) NOT NULL AUTO_INCREMENT,
  `name` varchar(255) CHARACTER SET utf8 COLLATE utf8_general_ci NOT NULL,
  `age` int(11) NOT NULL,
  PRIMARY KEY (`id`) USING BTREE
) ENGINE = InnoDB AUTO_INCREMENT = 1 CHARACTER SET = utf8 COLLATE = utf8_general_ci ROW_FORMAT = Dynamic;
```
此时表中主键有AutoIncrement标记，如下图所示，左下角的自动递增已经勾选。
image

2、使用"gorm.io/gen" 生成该该数据库对应的表结构
```go
g := gen.NewGenerator(gen.Config{		
		ModelPkgPath: "./impl/mysql/model",
		FieldNullable: true,
		FieldCoverable: true,
		FieldWithIndexTag: true,
		FieldWithTypeTag: true,
	})
	db, _ := gorm.Open(mysql.Open("root:lvlv@(127.0.0.1:3306)/example?charset=utf8mb4&parseTime=True&loc=Local"))
	g.UseDB(db)
	g.GenerateAllTable()
```
运行后生成的数据库表结构体如下：
```go
package model

const TableNamePerson = "people"

// Person mapped from table <people>
type Person struct {
	ID   int32  `gorm:"column:id;type:int(20);primaryKey;autoIncrement:true" json:"id"`
	Name string `gorm:"column:name;type:varchar(255);not null" json:"name"`
	Age  int32  `gorm:"column:age;type:int(11);not null" json:"age"`
}

// TableName Person's table name
func (*Person) TableName() string {
	return TableNamePerson
}

```

3、编写另外一个cmd程序，利用上述生成的struct 来自动建立数据库。
先新建数据库example1，在example1中利用上步生成的struct 重建Person表结构。
```go
        dsn := "root:lvlv@(127.0.0.1:3306)/example1?charset=utf8mb4&parseTime=True&loc=Local")
        db, err := gorm.Open(mysql.Open(dsn), &gorm.Config{
		SkipDefaultTransaction: true,
		NamingStrategy:         schema.NamingStrategy{SingularTable: true},		
	})
	e = db.AutoMigrate(&model.Person{})
	if e != nil {
		fmt.Println("AutoMigrate error!", e)
	}
```
此时生成的数据库example1中的表结构中对于主键id没有AutoIncrement选项。如下图所示，自动递增没有勾选。
image
这导致后续向该表中添加记录时，报错。

### PR DESCRIPTION
**Pull request by wclssdn**
add autoIncrement implement

close #5266

### CODE REVIEW

File: migrator/migrator.go [hunk 1]
```diff
@@ -73,6 +73,10 @@ func (m Migrator) FullDataTypeOf(field *schema.Field) (expr clause.Expr) {
 		expr.SQL += " NOT NULL"
 	}
 
+	if field.AutoIncrement {
+		expr.SQL += " AUTO_INCREMENT"
+	}
+
```
Comment (sunfuze) [inline migrator/migrator.go:79]: not suitable for postgres

Comment (wclssdn) [reply migrator/migrator.go:79]: I have not use that db, can you help me to implement that feature?

**Review by a631807682 — COMMENTED** [review]
You can use `docker-compose` to generate the test databases and fix them. You need to determine if this is a mysql driver problem or a problem with all drivers. 
1. If it is a problem with the mysql driver, you can create a PR for https://github.com/go-gorm/mysql
From the description, it looks like a problem with [AlterColumn](https://github.com/go-gorm/mysql/blob/master/migrator.go#L49), you can directly change or implement the `FullDataTypeOf` of the driver itself 
2. If it is a problem with all drivers, you do not need to implement this function in all drivers, but you need to ensure that they are not affected, and all tests need to pass.

**Comment by wclssdn** [comment]
It's too hard for me to implements all databases support. so, i think i should close this PR.

**Pull request closed without merging**